NitroxClient/GameLogic/Terrain.cs

```diff
@@ -77,11 +80,23 @@ public void UpdateVisibility()
 
     }
 
+    public static void WaitForEntities()
+    {
+        // In case the player is spawned in the air, we need to hold them up while all the entities load around them
+        if (Player.main && !Player.main.IsUnderwater() && !Player.main.groundMotor.grounded)
+        {
+            Player.main.cinematicModeActive = true;
+            CoroutineHost.StartCoroutine(WaitForWorldLoad(Yielders.WaitFor2Seconds));
+        }
+    }
+
     /// <summary>
     /// Forces world streamer's to load the terrain around the MainCamera and waits until it's done to unfreeze the player.
     /// </summary>
-    public static IEnumerator WaitForWorldLoad()
+    public static IEnumerator WaitForWorldLoad(WaitForSeconds initialWait = null)
     {
+        Entities entities = NitroxServiceLocator.LocateService<Entities>();
```

Comment:
You should be using dependency injection, this will generate a warning from our analyzers

Reply:
It won't work in this static context

Reply:
Terrain is already being required inside Mutiplayer mb, so static can be removed